lib/p2p/Pool.ts

```diff
@@ -135,6 +137,23 @@ class Pool extends EventEmitter {
     this.connected = false;
   }
 
+  private verifyReachability = () => {
+    this.handshakeData.addresses!.forEach(async (address) => {
+      const externalAddress = addressUtils.toString(address);
+      this.logger.debug(`Trying to connect to advertised address: ${externalAddress}`);
+      try {
+        const peer = Peer.fromOutbound(address, this.logger);
+        await peer.open(this.handshakeData, this.handshakeData.nodePubKey);
```

Comment:
perhaps mark somehow that this line should never pass without throwing exception. maybe add `assert(false)` after it.